src/SectorView.cpp

```diff
@@ -636,64 +649,74 @@ std::vector<SystemPath> SectorView::AutoRoute(const SystemPath &start, const Sys
 	// distance to the start is 0
 	path_dist[0] = 0.f;
 
+	size_t totalSkipped = 0u;
 	while (unvisited.size() > 0) {
 		// find the closest node (for the first loop this will be start)
 		std::vector<SystemPath>::size_type closest_i = *unvisited.begin();
-		for (auto it = unvisited.begin(); it != unvisited.end(); it++) {
+		for (auto it = unvisited.begin(), itEnd = unvisited.end(); it != itEnd; ++it) {
 			if (path_dist[*it] < path_dist[closest_i]) closest_i = *it;
 		}
-		SystemPath closest = nodes[closest_i];
-		RefCountedPtr<const Sector> closest_sec = m_galaxy->GetSector(closest);
 
 		// mark it as visited
 		unvisited.erase(closest_i);
 
 		// if this is the target then we have found the route
+		const SystemPath &closest = nodes[closest_i];
 		if (closest.IsSameSystem(target)) break;
 
+		RefCountedPtr<const Sector> closest_sec = m_galaxy->GetSector(closest);
+
 		// if not, loop through all unvisited nodes
 		// since every system is technically reachable from every other system
 		// everything is a neighbor :)
-		for (auto it = unvisited.begin(); it != unvisited.end(); it++) {
-			SystemPath v = nodes[*it];
-			RefCountedPtr<const Sector> v_sec = m_galaxy->GetSector(v);
+		for (auto it = unvisited.begin(), itEnd = unvisited.end(); it != itEnd; ++it) {
+			const auto idx = *it;
+			const SystemPath &v = nodes[idx];
+			if ((SystemPath::SectorDistance(closest, v)*Sector::SIZE) > max_range) {
+				++totalSkipped;
+				continue;
+			}
+
+			RefCountedPtr<const Sector> v_sec = m_galaxy->GetSector(v); // this causes it to generate a sector (slooooooow)
 
-			float v_dist_ly = 0;
-			v_dist_ly = Sector::DistanceBetween(closest_sec, closest.systemIndex, v_sec, v.systemIndex);
+			const float v_dist_ly = Sector::DistanceBetween(closest_sec, closest.systemIndex, v_sec, v.systemIndex);
 
 			// in this case, duration is used for the distance since that's what we are optimizing
 			float v_dist = hyperdrive.CallMethod<float>("GetDuration", Pi::player, v_dist_ly, max_range);
 
 			v_dist += path_dist[closest_i]; // we want the total duration from start to this node
-			if (v_dist < path_dist[*it]) {
+			if (v_dist < path_dist[idx]) {
 				// if our calculated duration is less than a previous value, this path is more efficent
 				// so store/override it
-				path_dist[*it] = v_dist;
-				path_prev[*it] = closest_i;
+				path_dist[idx] = v_dist;
+				path_prev[idx] = closest_i;
 			}
 		}
 	}
+	Output("SectorView::AutoRoute, total times that nodes were skipped = %lu\n", totalSkipped);
```

Comment:
Is that a leftover from the debug code? If yes, consider also removing totalSkipped.

Reply:
Deliberate, more info for the logs when people complain about it being slow.